src/lib/mongoBackend/mongoRegistrationGet.cpp

```diff
@@ -0,0 +1,325 @@
+
+  if (crV.size() > 1)
+  {
+    return false;
+  }
+
+  //
+  // Extract first CR from the contextRegistration vector
+  //
+  mongo::BSONObj cr0 = crV[0].embeddedObject();
+
+
+
+  //
+  // Entities
+  //
+  std::vector<mongo::BSONElement>  dbEntityV = getFieldF(cr0, REG_ENTITIES).Array();
+
+  for (unsigned int ix = 0; ix < dbEntityV.size(); ++ix)
+  {
+    ngsiv2::EntID    entity;
+    mongo::BSONObj   ce = dbEntityV[ix].embeddedObject();
+
+    if (ce.hasField(REG_ENTITY_ISPATTERN))
+    {
+      std::string isPattern = getStringFieldF(ce, REG_ENTITY_ISPATTERN);
+
+      if (isPattern == "true")
+      {
+        entity.idPattern = getStringFieldF(ce, REG_ENTITY_ID);
+      }
+      else
+      {
+        entity.id = getStringFieldF(ce, REG_ENTITY_ID);
+      }
+    }
+    else
+    {
+      entity.id = getStringFieldF(ce, REG_ENTITY_ID);
+    }
+    
+    if (ce.hasField(REG_ENTITY_ISTYPEPATTERN))
+    {
+      std::string isPattern = getStringFieldF(ce, REG_ENTITY_ISTYPEPATTERN);
+
+      if (isPattern == "true")
+      {
+        entity.typePattern = getStringFieldF(ce, REG_ENTITY_TYPE);
+      }
+      else
+      {
+        entity.type = getStringFieldF(ce, REG_ENTITY_TYPE);
+      }
+    }
+    else
+    {
+      entity.type = getStringFieldF(ce, REG_ENTITY_TYPE);
+    }
+
+    regP->dataProvided.entities.push_back(entity);
+  }
+
+
+  //
+  // Attributes
+  //
+  std::vector<mongo::BSONElement> dbAttributeV = getFieldF(cr0, REG_ATTRS).Array();
+  for (unsigned int ix = 0; ix < dbAttributeV.size(); ++ix)
+  {
+    mongo::BSONObj  aobj     = dbAttributeV[ix].embeddedObject();
+    std::string     attrName = getStringFieldF(aobj, REG_ATTRS_NAME);
+
+    if (attrName != "")
+    {
+      regP->dataProvided.attributes.push_back(attrName);
+    }
+  }
+
+  //
+  // Providing application
+  //
+  setProvider(regP, cr0);
+
+  return true;
+}
+
+
+
+/* ****************************************************************************
+*
+* setExpires -
+*/
+static void setExpires(ngsiv2::Registration* regP, const mongo::BSONObj& r)
+{
+  regP->expires = (r.hasField(REG_EXPIRATION))? getIntFieldF(r, REG_EXPIRATION) : -1;
+}
+
+
+
+/* ****************************************************************************
+*
+* setStatus -
+*/
+static void setStatus(ngsiv2::Registration* regP, const mongo::BSONObj& r)
+{
+  regP->status = (r.hasField(REG_STATUS))? getStringFieldF(r, REG_STATUS): "";
+}
+
+
+
+/* ****************************************************************************
+*
+* setForwardingInformation -
+*/
+static void setForwardingInformation(ngsiv2::Registration* regP, const mongo::BSONObj& r)
+{
+}
+
+
+
+/* ****************************************************************************
+*
+* mongoRegistrationGet - 
+*/
+void mongoRegistrationGet
+(
+  ngsiv2::Registration*  regP,
+  const std::string&     regId,
+  const std::string&     tenant,
+  const std::string&     servicePath,
+  OrionError*            oeP
```

Comment:
Is our "style" to have OrionError always at the end in mongoBackend "external" functions?

Reply:
We have no style about that, I'm afraid.
However, having this parameter last makes sense, at least to me.
First param is the receiver (output parameter), then the stuff to fill the output with, and the last param is part of the return value ...

I wouldn't want it to be mixed inside the input params, that would be strange.

Perhaps we should add something in the style guide ...

Reply:
Maybe adding some rule about "first all the input parameters, next all the output parameters" ?

Reply:
Thinking a little about this ...
Normally the output params would go first, look at strcpy for example:

`char* strcpy(char *dest, const char *src)` 

Not sure I like to put all output params first, but I really like to have the 'destination' first, if a function has a destination.

Perhaps a hybrid:
```
xxx func(dest, input1, .. inputN, output2, ... outputN).
```

Secondary output params are like bieffects ...

Perhaps better to leave this be.
I would though add a rule about if a function has a destination parameter, it should be the first parameter to the function. Either that, or nothing at all. Depends on the function ...

Reply:
If we distinguish two "categories" of output params (destination and "others") things start getting complicated and maybe the associated rule will be hard to follow. In this situation, maybe not having a rule at all... not sure.

I like the idea of having OrionError (or std::string error in some cases) at the end... the problem is that we don't follow that pattern in other cases. Could you create an issue about it, please? However, hard to detect "unaligned cases"...

Reply:
NTC for this PR. Another PR will deal with the style guide